evennia/contrib/currency.py

```diff
@@ -0,0 +1,374 @@
+"""
+Currency Module
+'Currency' classes represent monetary values on objects or characters.
+They are instantiated by the 'CurrencyHandler' object, which is typically
+set up as a property on the object or character's typeclass.
+
+**Setup**
+    To use currency on an object, add a function that passes the object
+    itself into the constructor and returns a 'CurrencyHandler'. This 
+    function should be decorated with the 'lazy_property' decorator.
+
+    Example:
+    ```python
+        from evennia.utils import lazy_property
+        from world.currency import CurrencyHandler
+
+        Class Object(DefaultObject):
+        ...
+        @lazy_property
+        def purse(self):
+            return CurrencyHandler(self, db_attribute='purse')
+    ```
+**Currency Configuration**
+    'Currency' objects can be configured with a name and a value (relative to other
+    Currency). All currency objects have a settable 'amt' property that contains the 
+    amount of that currency type.
+
+    Example:
+        ```python
+        >>> gp = obj.currency.gp
+        >>> gp.amt
+        100
+        ```
+
+    Constructor Args:
+        name(str): name of the currency type
+        value(int, float): value based on other currency types in handler
+        amt(int, float): amount of currency type held
+    
+    Methods:
+        convert(Currency, Optional amount): Convert between Currency types.
+        contents(): List all currencies that have amounts greater than zero
+        total(): Returns a total in lowest currency
+        
+    Examples:
+        ```python
+        >>> purse.CC.convert(self.purse.SC,2)
+        Converted 2 silver coin --> 20 copper coin
+        >>> purse.CC.amt = 4    # Add 4 copper coins

```

Comment:
Actually, `purse` is actually the name of the `CurrencyHandler` property on the object from the example above. So this should be `obj.purse`; it's not a stand-alone entity.